zerver/webhooks/gitlab/view.py

```diff
@@ -265,12 +265,29 @@ def get_pipeline_event_body(payload: Dict[str, Any]) -> str:
             project_homepage,
             build.get('id')
         )
-        builds_status += "* [{}]({}) - {}\n".format(
+        artifact_filename = build.get("artifacts_file").get("filename")
+        if artifact_filename:
+            artifact_download_url = '{}/artifacts/download'.format(build_url)
+            artifact_browse_url = '{}/artifacts/browse'.format(build_url)
+            artifact_string = '  * built artifact: *{}* [[Browse]({})|[Download]({})]\n'.format(
+                artifact_filename,
+                artifact_browse_url,
+                artifact_download_url
+            )
+        else:
+            artifact_string = ''
+        builds_status += "* [{}]({}) - {}\n{}".format(
             build.get('name'),
             build_url,
-            build.get('status')
+            build.get('status'),
+            artifact_string
         )
-    return "[Pipeline]({}) {} with build(s):\n{}.".format(pipeline_url, action, builds_status[:-1])
+    return "[Pipeline ({})]({}) {} with build(s):\n{}.".format(
+        payload['object_attributes'].get('id'),
```

Comment:
Can you make this `payload['object_attributes']['id']`?

Reply:
I tried to keep that coherent with some other lines (not mine) inside `view.py`:

https://github.com/zulip/zulip/blob/20ffce52389075e3aa10affd57aea80b89983a91/zerver/webhooks/gitlab/view.py#L160

https://github.com/zulip/zulip/blob/20ffce52389075e3aa10affd57aea80b89983a91/zerver/webhooks/gitlab/view.py#L215

I have not found any reference to an `id` field of a dict via square brackets and thought it was done to be error-prone.
Are you still voting for `payload['object_attributes']['id']`? Then I'll sure change it.

Reply:
Oh, I see. Guess I didn't notice that earlier. Good point. Let's keep this line as it is then. :slightly_smiling_face:

Reply:
You can mark this comment as resolved now (by hitting that button somewhere near the start of this comment thread).